test/js/web/fetch/fetch.test.ts

```diff
@@ -1596,3 +1596,44 @@ it("same-origin status code 302 should not strip headers", async () => {
   expect(redirected).toBe(true);
   server.stop(true);
 });
+
+it.each([
+  ["/a/b", "/c", "/c"],
+  ["/a/b", "c", "/a/c"],
+  ["/a/b", "/c/d", "/c/d"],
+  ["/a/b", "c/d", "/a/c/d"],
+  ["/a/b", "../c", "/c"],
+  ["/a/b", "../c/d", "/c/d"],
+  ["/a/b", "../../../c", "/c"],
+  // slash
+  ["/a/b/", "/c", "/c"],
+  ["/a/b/", "c", "/a/b/c"],
+  ["/a/b/", "/c/d", "/c/d"],
+  ["/a/b/", "c/d", "/a/b/c/d"],
+  ["/a/b/", "../c", "/a/c"],
+  ["/a/b/", "../c/d", "/a/c/d"],
+  ["/a/b/", "../../../c", "/c"],
+])("should handle relative location in the redirect ('%s', '%s'), issue#5635", async (pathname, location, expected) => {
+  const server = Bun.serve({
```

Comment:
This test will be kind of slow/unreliable due to occupying a unique port per call 

Instead, can we create a single Bun.serve() instance, call `.reload` and then one .stop(true) inside an afterAll?

Reply:
Fixed.